### DISCUSSION TITLE
Feature: Benchmark/performance information

### DISCUSSION THREAD

**Discussion by MeijeSibbel** [siacentral/host-manager#101] [Feature Requests]
Although i can see that my host is storing data and technically working, i would like to know how well the host  is working (e.g. latency, up-down speed) and maybe compare it to other hosts through e.g. benchmarking. Sia Stats for example does:

![image](https://user-images.githubusercontent.com/11735738/64412535-dab97a00-d08f-11e9-82a7-34df3a8db2be.png)

Adding some of those scores/metrics to the dashboard will enhance the person's understanding of how good their Sia host actually a=performs.

**Comment by n8mgr** [comment]
I like this idea a lot and it's something I've been trying to look into, not just for the Host Manager. My only concern is metrics like that can be really problematic. 

SiaStats has problems with the latency of hosts in different locations as well as the small size of the uploaded and downloaded data not having enough time to fully saturate the connection or accurately depict speeds. Adding more locations or increasing the size of the test data increases the cost of the benchmarks.

If you have any thoughts or ideas I'm open to them

A distributed public hostdb would be one way to solve this, but horribly difficult to implement and not very trustworthy.

**Comment by MeijeSibbel** [comment]
How about getting some of these metrics client-side?. Upload and download speed can be determined on the client post-transfer by taking the size of the chunk and the time to transfer it. Although this also depends on the pipe of the renter, with enough chunks a pretty good picture can be created. Not sure if this is possible for latency however since that one depends on the difference in time the host receives the request and opens the upload/download channel.

**Comment by n8mgr** [comment]
The only problem I see with that is Sia does not give us any access to upload/download speed stats or even when a chunk is being actively transferred. I could work on a pull request to Sia core to add stats like that, I'm sure it'd be welcome
